thorlcr/thorutil/thmem.cpp

```diff
@@ -229,6 +243,8 @@ class CSpillableStreamBase : public CSimpleInterface, implements roxiemem::IBuff
     }
     virtual bool freeBufferedRows(bool critical)
     {
+        if (spillFile) // if spilt already. NB: thread-safe, as only set by this thread and can't this freeBufferedRows() can't be called on multiple threads concurrently.
+            return false;
```

Comment:
Seems a bit of a strange test - is it not possible that you need to spill again once you have spilled once?